modules/integration/tests-integration/tests-security-manager/src/test/java/org/wso2/carbon/appserver/integration/test/server/security/manager/Carbon3340TestCase.java

```diff
@@ -0,0 +1,141 @@
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.appserver.integration.common.clients.WebAppAdminClient;
+import org.wso2.appserver.integration.common.utils.ASIntegrationTest;
+import org.wso2.appserver.integration.common.utils.WebAppDeploymentUtil;
+import org.wso2.appserver.integration.common.utils.WebAppTypes;
+import org.wso2.carbon.automation.engine.FrameworkConstants;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.exceptions.AutomationFrameworkException;
+import org.wso2.carbon.automation.engine.frameworkutils.enums.OperatingSystems;
+import org.wso2.carbon.automation.test.utils.common.TestConfigurationProvider;
+import org.wso2.carbon.integration.common.admin.client.UserManagementClient;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test class to verify the ability to invoke user store operations from deployed
+ * webapps when java security manager is enabled.
+ */
+
+public class Carbon3340TestCase extends ASIntegrationTest {
+
+
+    private static final Log log = LogFactory.getLog(Carbon3340TestCase.class);
+
+    private WebAppAdminClient webAppAdminClient;
+    private String webAppUrl;
+    private final String hostName = "localhost";
+    private final String webAppFileName = "carboncontext-test-app.war";
+    private final String webAppName = "carboncontext-test-app";
+
+    private String user1 = "UserStoreTestUser1";
+    private String user2 = "UserStoreTestUser2";
+
+    private TestUserMode userMode;
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public Carbon3340TestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+        };
+    }
+
+    @BeforeClass(alwaysRun = true, description = "Deploying the Web application")
+    public void init() throws Exception {
+        if (System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_OS_NAME).toLowerCase()
+                .contains(OperatingSystems.WINDOWS.toString().toLowerCase())) {
+            throw new SkipException("Skipping this test case in windows");
+        }
+        super.init(userMode);
+        webAppAdminClient = new WebAppAdminClient(backendURL, sessionCookie);
+        webAppAdminClient.uploadWarFile(TestConfigurationProvider.getResourceLocation("AS")
+                + File.separator + "security"
+                + File.separator + "manager" + File.separator + "webapp"
+                + File.separator + webAppFileName);
+        //let webapp to deploy
+        Thread.sleep(2000);
```

Comment:
Better to use WebAppDeploymentUtil.isWebApplicationDeployed() instead of the Thread.sleep()